core/src/mender-zephyr-image-update-module.c

```diff
@@ -29,6 +29,8 @@
  */
 static void *mcu_boot_flash_handle = NULL;
 
+static bool artifact_has_payload;
```

Comment:
I'd actually name this `artifact_had_payload` as that reflects that it's only valid after an artifact was processed.